flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/RexNodeTranslator.scala

```diff
@@ -74,15 +75,52 @@ object RexNodeTranslator {
 
   /**
     * Parses all input expressions to [[UnresolvedAlias]].
-    * And expands star to parent's full project list.
+    * And expands star to parent's full project list and flattens composite types.
     */
-  def expandProjectList(exprs: Seq[Expression], parent: LogicalNode): Seq[NamedExpression] = {
+  def expandProjectList(
+      exprs: Seq[Expression],
+      parent: LogicalNode,
+      tableEnv: TableEnvironment)
+    : Seq[NamedExpression] = {
+
     val projectList = new ListBuffer[NamedExpression]
     exprs.foreach {
+
       case n: UnresolvedFieldReference if n.name == "*" =>
         projectList ++= parent.output.map(UnresolvedAlias(_))
+
+      // flattening can only applied on field references
+      case Flattening(composite) if
```

Comment:
There is already a test for that in the last line of `CompositeAccessTest.testGetField`. The flattening on a flat type simply has no effect at all.